datafusion/optimizer/src/simplify_expressions/simplify_exprs.rs

```diff
@@ -505,59 +488,6 @@ mod tests {
         Ok(())
     }
 
-    #[test]
-    fn now_less_than_timestamp() -> Result<()> {
```

Comment:
Moved to datafusion/core/tests/simplification.rs

Reply:
There are in fact two problems with the commented out code.

1. call_fn only works for BuiltinScalarFunction. Easy to work around by using a direct function call (once the test is moved)
2. test_evaluate calls test_evaluate_with_start_time which builds a ConstEvaluator directly. Calling the entire simplifier on them should fix this issue - the fix is really in test_evaluate_with_start_time function (when moved). Moving that will basically mean all the test_evaluate functions would be moved over as well.

Reply:
That sounds reasonable. Let me know if you need help moving the code / tests for with start time

Reply:
Migrated with a comment ([see below](https://github.com/apache/arrow-datafusion/pull/9388#discussion_r1506948609))